src/olympia/blocklist/mlbf.py

```diff
@@ -168,21 +192,18 @@ def soft_blocked_items(self) -> List[str]:
 
     @cached_property
     def not_blocked_items(self) -> List[str]:
-        all_blocks_ids = [version.version_id for version in self._all_blocks]
+        all_blocks_ids = [version[2] for version in self._all_blocks]
         not_blocked_items = MLBF.hash_filter_inputs(
-            Version.unfiltered.exclude(id__in=all_blocks_ids or ())
-            .distinct()
-            .order_by('id')
-            .values_list('addon__addonguid__guid', 'version')
+            get_not_blocked_items(all_blocks_ids)
         )
+        blocked_items = set(self.blocked_items + self.soft_blocked_items)
         # even though we exclude all the version ids in the query there's an
         # edge case where the version string occurs twice for an addon so we
         # ensure not_blocked_items contain no blocked_items or soft_blocked_items.
-        return [
-            item
-            for item in not_blocked_items
-            if item not in self.blocked_items + self.soft_blocked_items
-        ]
+        newly_not_blocked_items, _, _ = ordered_diff_lists(
```

Comment:
I get why you're reusing this but is it necessary? vs. `return [item for item not in not_blocked_items if item not is blocked_items]`? isn't `ordered_diff_lists` computing two lists?

Reply:
I don't see the issue reusing it. it's doing the exact same thing and better ensures that we use the most performant loop. Had we already been using that method, we wouldn't have needed this patch at all.

Reply:
isn't `ordered_diff_lists` computing two lists?

Reply:
Yes, and that's what we want. we have `blocked_items` and `not_blocked_items` and we want to diff them. ordered_diff_lists returns 3 values `added`, `removed`, `count`. We want added, so we return that value.

Reply:
right, and what I am saying is that we could do without the computation of `removed`, no?

Reply:
we can move this conversation elsewhere if the method call moves alongside. I don't disagree that it's a can vs. should situation. Since we're trying to fix prod, maybe we should change the minimum amount of code?

Reply:
@willdurand this code passed the test, we are going to verify it on stage. It's arguably more code change to not reuse the existing tested, verified method. I don't want to change this any more than I already have.

Reply:
I am not clear on why you insist on doing something here that is more code changes while - at the same time - changing the code in https://github.com/mozilla/addons-server/pull/22861 to do exactly what I've been suggesting so far. It sounds to me that this is the simplest diff we can come up with:

```diff
diff --git a/src/olympia/blocklist/mlbf.py b/src/olympia/blocklist/mlbf.py
index d2cc578afb..fa3a742f7b 100644
--- a/src/olympia/blocklist/mlbf.py
+++ b/src/olympia/blocklist/mlbf.py
@@ -175,14 +175,11 @@ class MLBFDataBaseLoader(BaseMLBFLoader):
             .order_by('id')
             .values_list('addon__addonguid__guid', 'version')
         )
+        blocked_items = set(self.blocked_items + self.soft_blocked_items)
         # even though we exclude all the version ids in the query there's an
         # edge case where the version string occurs twice for an addon so we
         # ensure not_blocked_items contain no blocked_items or soft_blocked_items.
-        return [
-            item
-            for item in not_blocked_items
-            if item not in self.blocked_items + self.soft_blocked_items
-        ]
+        return [item for item in not_blocked_items if item not in blocked_items]
```

I'll add that statements like "I don't want to change this any more than I already have" don't belong to code review discussions. This is a conversation where we both need to be convinced that we got to the best solution (for some definition of "best").

Reply:
Done.

Reply:
Fwiw I did not update the second PR to what you are suggesting but to a third option which addresses your concern and maintains my (legitimate) goal of re-using the single function for diffing.